app/models/cloud_database.rb

```diff
@@ -10,8 +10,41 @@ class CloudDatabase < ApplicationRecord
 
   serialize :extra_attributes
 
+  supports_not :create
   supports_not :delete
 
+  def self.create_cloud_database_queue(userid, ext_management_system, options = {})
+    task_opts = {
+      :action => "creating Cloud Database for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => 'CloudDatabase',
+      :method_name => 'create_cloud_database',
+      :role        => 'ems_operations',
+      :queue_name  => ext_management_system.queue_name_for_ems_operations,
+      :zone        => ext_management_system.my_zone,
+      :args        => [ext_management_system.id, options]
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def self.create_cloud_database(ems_id, options = {})
+    raise ArgumentError, _("ems_id cannot be nil") if ems_id.nil?
+
+    ext_management_system = ExtManagementSystem.find(ems_id)
+    raise ArgumentError, _("ext_management_system cannot be found") if ext_management_system.nil?
```

Comment:
`.find()` will raise a record not found exception if it can't find it so you don't need the extra check herw